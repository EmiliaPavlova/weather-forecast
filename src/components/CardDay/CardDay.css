.card_title {
  font-size: 1.6rem;
}

.card_icon {
  display: flex;
  margin: 0 auto;
}