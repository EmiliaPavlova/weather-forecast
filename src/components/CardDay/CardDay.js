import React from 'react';
import { convertToCelsius } from '../../Utils/utils';
import './CardDay.css';

export default function CardDay({
  date,
  day,
  icon,
  weather,
  description,
  temp,
  feelsLike,
  tempMin,
  tempMax,
}) {

  const iconLink = `http://openweathermap.org/img/wn/${icon}@2x.png`;

  return (
    <>
      <div className=''>{date}</div>
      <div className='card_title'>{day}</div>
      <img className='card_icon' src={iconLink} alt={description}/>
      <div className='card_title'>{weather}, {convertToCelsius(temp)}&#8451;</div>
      <div className=''>Feels like {convertToCelsius(feelsLike)}&#8451;</div>
      <div className=''>{convertToCelsius(tempMin)}&#8451; - {convertToCelsius(tempMax)}&#8451;</div>
    </>
  )
};