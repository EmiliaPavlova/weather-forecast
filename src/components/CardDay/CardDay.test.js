import React from 'react';
import { shallow } from 'enzyme';
import CardDay from './CardDay';

describe('CardDay component', () => {
  it('converts temp props from Kelvin to Celsius', () => {
    const wrapper = shallow(<CardDay
      date='15 April'
      day='Friday'
      icon='icon'
      weather='Sunny'
      description='sunny with clouds'
      temp={292.15}
      feelsLike={292.15}
      tempMin={279.15}
      tempMax={297.15}
    />);
    expect(wrapper).toMatchSnapshot();
  })
})