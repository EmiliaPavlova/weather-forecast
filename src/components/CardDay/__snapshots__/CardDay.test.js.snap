// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDay component converts temp props from Kelvin to Celsius 1`] = `
<Fragment>
  <div
    className=""
  >
    15 April
  </div>
  <div
    className="card_title"
  >
    Friday
  </div>
  <img
    alt="sunny with clouds"
    className="card_icon"
    src="http://openweathermap.org/img/wn/icon@2x.png"
  />
  <div
    className="card_title"
  >
    Sunny
    , 
    19
    ℃
  </div>
  <div
    className=""
  >
    Feels like 
    19
    ℃
  </div>
  <div
    className=""
  >
    6
    ℃ - 
    24
    ℃
  </div>
</Fragment>
`;
